metabase/metabase
# Fix - Trend charts look very broken with new dashboard grid changes

Closes #31628

### Description

This fixes overflow in:
- simple scalar charts (screenshots: [before](https://github.com/metabase/metabase/assets/6830683/d5d121fb-8f0d-4bce-a4e7-955ae6545256) / [after](https://github.com/metabase/metabase/assets/6830683/1d46ff18-76d7-4666-8f41-b6cac3eb68ab))
- trend charts (screenshots: [before](https://github.com/metabase/metabase/assets/6830683/fba4ee07-a93d-4aaf-bb84-f37fe6bcfcc0) / [after](https://github.com/metabase/metabase/assets/6830683/3125ac69-78a6-4901-8014-15ca345552b2))

It works down to 2x2 size in `smartscalar`, so this PR also updates the minimum allowed size of `smartscalar`.

To see how the regression looked like please [watch these recordings](https://github.com/metabase/metabase/issues/31628#issuecomment-1597120018).

Related Slack threads:
- https://metaboat.slack.com/archives/C057T1QTB3L/p1687167736234729
- https://metaboat.slack.com/archives/C505ZNNH4/p1687463683427169

### Tests

I went with e2e tests because in unit tests elements don't have dimensions and mocking the visual state that we expect would defy the purpose.

### How to verify

Create a dashboard and add a bunch of questions visualized as: "number" and "trend" (here's the question for the trend chart I'm using in my recordings: [screenshot](https://github.com/metabase/metabase/assets/6830683/d14143a9-61c2-4b6f-8bfe-31001484f523)).

The following things affect how the cards are displayed, please try various combinations:
- card size
- viewport size (browser window size)
- whether left-hand side navigation sidebar is open or not
- whether the card has a description or not (there will be an "i" icon displayed)
- size of the text values displayed in the card (card value, card name, trend difference, etc.)

Less important but also affecting:
- font (user setting)
- language (user setting)

How it's meant to work:
- card title will be displayed in 2 lines if possible, in 1 line otherwise
- card value & title will be truncated with ellipsis if there's not enough horizontal space
- card value font size is automatically computed based on the available horizontal & vertical space (there is a min & max to this)
- if something is truncated with ellipsis, a tooltip with a complete value should be shown on hover
- if something is not truncated with ellipsis, there should be no tooltip
- description icon is shown only when description is present and description is shown in tooltip on hover
- in trend cards:
  - the `"was XYZ last month"` label will be displayed only if it fits horizontally - otherwise it will be shown in tooltip on hover
    - except on mobile resolutions, where it will be displayed in a separate row
  - the % difference will be always shown
    - it will try to use up to 2 decimal points precision if possible - otherwise it will try to use less precision
    - if there's not enough room to show it anyway - it will be truncated
    - icon is always shown

### Demo


#### Trend - Before

https://github.com/metabase/metabase/assets/6830683/5ffbd7a2-f95a-4e40-90b9-79656751330b

#### Trend - After

https://github.com/metabase/metabase/assets/6830683/8ed64ce7-baaf-48a6-8274-b81114f3b41c

#### Number - Before

https://github.com/metabase/metabase/assets/6830683/e06da892-6e07-4bd4-8cb1-51d699fe43c3




#### Number - After

https://github.com/metabase/metabase/assets/6830683/0c41745f-fb86-4fb7-b193-bcd642d678ec
